app/src/main/java/com/github/h3lp3rs/h3lp/forum/ForumCategory.kt

```diff
@@ -23,17 +29,31 @@ enum class ForumCategory {
         fun forumOf(choice: ForumCategory): Forum {
             // The categories are the first sub-level of the forum, thus they lie after root (the
             // empty list)
-            choice.forum = choice.forum ?: FireForum(listOf(choice.name))
-            return choice.forum!!
+            root?.let {
+                return it.child(choice.name)
+            }
+            root = FireDBForum(emptyList(), FireDatabase(ROOT_FORUM_DB_PATH))
+            return root!!.child(choice.name)
         }
 
         /**
-         * Used for testing purposes to give forum instances
-         * @param newForum The forum to use
+         * Returns the root of the forum
          */
-        fun setForum(choice: ForumCategory, newForum: Forum) {
-            choice.forum = newForum
+        fun root(): Forum {
+            root?.let {
+                return it
+            }
+            root = FireDBForum(emptyList(), FireDatabase(ROOT_FORUM_DB_PATH))
+            return root!!

```

Comment:
same idea as above to avoid the "bad practice" of !!
```suggestion
            rootForum = FireDBForum(emptyList(), FireDatabase(ROOT_FORUM_DB_PATH))
            root = rootForum
            return rootForum
```

Reply:
I want to push this, but I won't waste a whole Cirrus build for this.